### PR TITLE
[helm] add the ability for the module to uninstall pending-install releases

### LINKED ISSUE #319: kubernetes.core.helm does not uninstall charts in pending-install status
##### SUMMARY
The module kubernetes.core.helm does not uninstall charts that are in a pending-install, pending-upgrade or similar STATUS

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
kubernetes.core.helm

##### ANSIBLE VERSION

```
ansible 2.10.11
  config file = /stratio/ansible/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.8/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.10 (default, May  6 2021, 00:05:59) [GCC 10.2.1 20201203]

```

##### COLLECTION VERSION

```
kubernetes.core    2.0.2  
community.kubernetes      1.1.1
```

##### CONFIGURATION

```
ANSIBLE_PIPELINING(/stratio/ansible/ansible.cfg) = True
ANSIBLE_SSH_ARGS(/stratio/ansible/ansible.cfg) = -o ControlMaster=auto -o ControlPersist=30m
ANSIBLE_SSH_RETRIES(/stratio/ansible/ansible.cfg) = 3
ANY_ERRORS_FATAL(/stratio/ansible/ansible.cfg) = True
CACHE_PLUGIN(/stratio/ansible/ansible.cfg) = jsonfile
CACHE_PLUGIN_CONNECTION(/stratio/ansible/ansible.cfg) = /tmp/facts_cache
CACHE_PLUGIN_TIMEOUT(/stratio/ansible/ansible.cfg) = 7200
COLLECTIONS_PATHS(/stratio/ansible/ansible.cfg) = ['/stratio/ansible/collections']
DEFAULT_BECOME(/stratio/ansible/ansible.cfg) = True
DEFAULT_CALLBACK_PLUGIN_PATH(env: ANSIBLE_CALLBACK_PLUGINS) = ['/stratio/ansible/plugins/callback']
DEFAULT_CALLBACK_WHITELIST(/stratio/ansible/ansible.cfg) = ['keos_profile_roles', 'timer']
DEFAULT_FORKS(/stratio/ansible/ansible.cfg) = 30
DEFAULT_GATHERING(/stratio/ansible/ansible.cfg) = smart
DEFAULT_HOST_LIST(env: ANSIBLE_INVENTORY) = ['/stratio/ansible/inventory']
DEFAULT_ROLES_PATH(/stratio/ansible/ansible.cfg) = ['/stratio/ansible/roles', '/stratio/ansible/playbooks/kubespray/roles']
DEFAULT_STDOUT_CALLBACK(/stratio/ansible/ansible.cfg) = yaml
DEFAULT_TIMEOUT(/stratio/ansible/ansible.cfg) = 30
DEFAULT_VAULT_PASSWORD_FILE(env: ANSIBLE_VAULT_PASSWORD_FILE) = /root/.credentials
DEPRECATION_WARNINGS(/stratio/ansible/ansible.cfg) = False
TRANSFORM_INVALID_GROUP_CHARS(/stratio/ansible/ansible.cfg) = ignore

```

##### OS / ENVIRONMENT

NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.13.7
PRETTY_NAME="Alpine Linux v3.13"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://bugs.alpinelinux.org/"

##### STEPS TO REPRODUCE

```yaml
<15:05:15> [keos-installer:0.4.0-SNAPSHOT@keosdev2] [/workspace] # helm list -n keos-ops -a
NAME         	NAMESPACE	REVISION	UPDATED                                	STATUS         	CHART              	APP VERSION
keos-operator	keos-ops 	1       	2021-12-29 14:55:01.664330975 +0000 UTC	pending-install	keos-operator-0.4.0	0.4.0      
(reverse-i-search)`': ^C
<15:05:25> [keos-installer:0.4.0-SNAPSHOT@keosdev2] [/workspace] 130# cat /tmp/lala.yaml 
- hosts: localhost
  tasks:
  - name: "Uninstall keos-operator helm chart if its status is not deployed"
    kubernetes.core.helm:
      name: keos-operator
      release_namespace: keos-ops
      release_state: absent


  - name: Deploy keos-operator helm chart
    kubernetes.core.helm:
      name: "keos-operator"
      chart_ref: "/stratio/helm/keos-operator"
      release_namespace: "keos-ops"

<15:05:27> [keos-installer:0.4.0-SNAPSHOT@keosdev2] [/workspace] # ansible-playbook /tmp/lala.yaml 

PLAY [localhost] *****************************************************************************************************************************************************************************************************************************************************************

TASK [Uninstall keos-operator helm chart if its status is not deployed] **********************************************************************************************************************************************************************************************************
» Wednesday 29 December 2021  15:05:35 +0000 (0:00:00.197)       0:00:00.197 *** 
ok: [localhost]

TASK [Deploy keos-operator helm chart] *******************************************************************************************************************************************************************************************************************************************
» Wednesday 29 December 2021  15:05:35 +0000 (0:00:00.613)       0:00:00.810 *** 
fatal: [localhost]: FAILED! => changed=false 
  command: /usr/local/bin/helm upgrade -i --reset-values keos-operator /stratio/helm/keos-operator
  msg: |-
    Failure when executing Helm command. Exited 1.
    stdout:
    stderr: Error: UPGRADE FAILED: "keos-operator" has no deployed releases
  stderr: |-
    Error: UPGRADE FAILED: "keos-operator" has no deployed releases
  stderr_lines: <omitted>
  stdout: ''
  stdout_lines: <omitted>

NO MORE HOSTS LEFT ***************************************************************************************************************************************************************************************************************************************************************

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

» Wednesday 29 December 2021  15:05:36 +0000 (0:00:00.740)       0:00:01.550 *** 
=============================================================================== 
kubernetes.core.helm ---------------------------------------------------- 1.35s
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
total ------------------------------------------------------------------- 1.35s
Playbook run took 0 days, 0 hours, 0 minutes, 1 seconds

```

##### EXPECTED RESULTS

Uninstall desired chart

##### ACTUAL RESULTS

Chart stays as pending-install, pending-upgrade or similar, and uninstall gives you and OK result.

```paste below

```

### PR DESCRIPTION
**Pull request by abikouo**
##### SUMMARY

closes #319 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

`helm`

### CODE REVIEW

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

File: changelogs/fragments/589-helm-uninstall-chart-releases-with-statuses-different-than-deployed.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
```
Comment (gravesm) [inline changelogs/fragments/589-helm-uninstall-chart-releases-with-statuses-different-than-deployed.yaml:2]: I think this should actually be a bugfix.

**Comment by abikouo** [comment]
recheck

**Review by gravesm — APPROVED** [review]

**Comment by abikouo** [comment]
regate

**Pull request merged**